# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: third_party/rust/ron/tests/numbers.rs
# Commit: 3f167f8b3137
# Full Hash: 3f167f8b313749f50f135e189e5854bef7ac0c73
# Author: Erich Gubler <erichdongubler@gmail.com>
# Date: 2025-10-03 18:08 +0000
# Regressor Bug: 1991551
# File Overlap Count: 23
# Overlapping Files: third_party/rust/wgpu-hal/Cargo.toml, supply-chain/audits.toml, third_party/rust/naga/src/back/spv/writer.rs, third_party/rust/naga/src/back/msl/writer.rs, third_party/rust/wgpu-types/src/lib.rs
# Description:
#   Bug 1991551 - build(webgpu): update WGPU to 8c4aebc0c1b75e3f7412dfc180745a0717fc30ec r=webgpu-reviewers,supply-chain-reviewers,nical
# ==============================================================================

diff -r 823439f307a0 -r 3f167f8b3137 third_party/rust/ron/tests/numbers.rs
--- a/third_party/rust/ron/tests/numbers.rs	Fri Oct 03 18:01:30 2025 +0000
+++ b/third_party/rust/ron/tests/numbers.rs	Fri Oct 03 18:08:29 2025 +0000
@@ -1,6 +1,6 @@
 use ron::{
     de::from_str,
-    error::{Error, Position, SpannedError},
+    error::{Error, Position, Span, SpannedError},
 };
 
 #[test]
@@ -13,21 +13,30 @@
         from_str::<u8>("0x"),
         Err(SpannedError {
             code: Error::ExpectedInteger,
-            position: Position { line: 1, col: 3 },
+            span: Span {
+                start: Position { line: 1, col: 1 },
+                end: Position { line: 1, col: 3 },
+            }
         })
     );
     assert_eq!(
         from_str::<u8>("0x_1"),
         Err(SpannedError {
             code: Error::UnderscoreAtBeginning,
-            position: Position { line: 1, col: 3 },
+            span: Span {
+                start: Position { line: 1, col: 1 },
+                end: Position { line: 1, col: 3 },
+            }
         })
     );
     assert_eq!(
         from_str::<u8>("0xFFF"),
         Err(SpannedError {
             code: Error::IntegerOutOfBounds,
-            position: Position { line: 1, col: 6 },
+            span: Span {
+                start: Position { line: 1, col: 3 },
+                end: Position { line: 1, col: 6 },
+            }
         })
     );
 }
@@ -42,21 +51,30 @@
         from_str::<u8>("0b"),
         Err(SpannedError {
             code: Error::ExpectedInteger,
-            position: Position { line: 1, col: 3 },
+            span: Span {
+                start: Position { line: 1, col: 1 },
+                end: Position { line: 1, col: 3 },
+            }
         })
     );
     assert_eq!(
         from_str::<u8>("0b_1"),
         Err(SpannedError {
             code: Error::UnderscoreAtBeginning,
-            position: Position { line: 1, col: 3 },
+            span: Span {
+                start: Position { line: 1, col: 1 },
+                end: Position { line: 1, col: 3 },
+            }
         })
     );
     assert_eq!(
         from_str::<u8>("0b111111111"),
         Err(SpannedError {
             code: Error::IntegerOutOfBounds,
-            position: Position { line: 1, col: 12 },
+            span: Span {
+                start: Position { line: 1, col: 3 },
+                end: Position { line: 1, col: 12 },
+            }
         })
     );
 }
@@ -71,21 +89,30 @@
         from_str::<u8>("0o"),
         Err(SpannedError {
             code: Error::ExpectedInteger,
-            position: Position { line: 1, col: 3 },
+            span: Span {
+                start: Position { line: 1, col: 1 },
+                end: Position { line: 1, col: 3 },
+            }
         })
     );
     assert_eq!(
         from_str::<u8>("0o_1"),
         Err(SpannedError {
             code: Error::UnderscoreAtBeginning,
-            position: Position { line: 1, col: 3 },
+            span: Span {
+                start: Position { line: 1, col: 1 },
+                end: Position { line: 1, col: 3 },
+            }
         })
     );
     assert_eq!(
         from_str::<u8>("0o77777"),
         Err(SpannedError {
             code: Error::IntegerOutOfBounds,
-            position: Position { line: 1, col: 8 },
+            span: Span {
+                start: Position { line: 1, col: 3 },
+                end: Position { line: 1, col: 8 },
+            }
         })
     );
 }
@@ -100,14 +127,20 @@
         from_str::<i8>("-_1"),
         Err(SpannedError {
             code: Error::UnderscoreAtBeginning,
-            position: Position { line: 1, col: 2 },
+            span: Span {
+                start: Position { line: 1, col: 1 },
+                end: Position { line: 1, col: 2 },
+            }
         })
     );
     assert_eq!(
         from_str::<u8>("256"),
         Err(SpannedError {
             code: Error::IntegerOutOfBounds,
-            position: Position { line: 1, col: 4 },
+            span: Span {
+                start: Position { line: 1, col: 1 },
+                end: Position { line: 1, col: 4 },
+            }
         })
     );
 }